### ISSUE TITLE
Wireguard UI doesnt restart Wireguard Service When Applying Config

### ISSUE THREAD

**Issue by fattyredpanda** [ngoduykhanh/wireguard-ui#478]
Wireguard UI doesnt restart Wireguard Service When Applying Config. I have to manually restart from console everytime after clicking apply config for it to take effect.

**Comment by alex85us** [comment]
+1

**Comment by vrtareg** [comment]
Same here
Trying to run on TrueNAS FreeBSD based Jail
Config is generated OK but unable to restart service and get stats of clients.

**Comment by thenktor** [comment]
Same on FreeBSD. To restart the service it should run `service wireguard restart`

**Comment by vgallegoiz** [comment]
Same as you, to apply the config I need to restart the wireguard container and then the wireguard-ui. Easier than `service wireguard restart`

**Comment by OlegMuzych** [comment]
Hi guys. To automatically restart the service, you need to: WGUI_MANAGE_RESTART=true
https://github.com/ngoduykhanh/wireguard-ui#using-docker

**Comment by dvett01** [comment]
I have issues with the WGUI_MANAGE_RESTART=true.
Once I hit apply config my wireguard tunnell will be taken down, but will not start again.

I believe it might be related to an issue with iptables that I can see in my docker logs:
ip link delete dev wg0
iptables -D FORWARD -i wg0 -j ACCEPT; iptables -t nat -D POSTROUTING -o ens6 -j MASQUERADE
/usr/bin/wg-quick: line 295: iptables: command not found

To bring wireguard up again I need to run it manually with wg-quick start wg0

Any idea if I am doing something wrong?

**Comment by kaganae** [comment]
I am having the same issue, running within docker on a Synololgy NAS. When I apply the config, the changes are applied to wg0.conf, the wg0 interface is stopped, but it does not start after (but works fine when I manually start from the command line via wg-quick). Environment variables are as follows:

environment:
      - SENDGRID_API_KEY
      - EMAIL_FROM_ADDRESS
      - EMAIL_FROM_NAME
      - SESSION_SECRET
      - WG_CONF_TEMPLATE
      - WGUI_MANAGE_START=false
      - WGUI_MANAGE_RESTART=true

**Comment by jantschaefer** [comment]
I'm having the same problem on my Synology. #520 did not solve the issuse for me. Did anyone else already try version 0.6.0 and can confirm that the problem still persits?

**Comment by bukiru** [comment]
> Hi guys. To automatically restart the service, you need to: WGUI_MANAGE_RESTART=true https://github.com/ngoduykhanh/wireguard-ui#using-docker

but I don't use docker!!!!

**Comment by kaganae** [comment]
Yes, thanks. WGUI_MANAGE_RESTART is set to true, but it isn't working as documented.

**Comment by KyKyIIIKuH** [comment]
Replaced this
ExecStart=/usr/bin/systemctl restart wg-quick@wg0.service
on this
ExecStart=/usr/bin/systemctl reload wg-quick@wg0.service

and everything started working
